### ISSUE TITLE
Possible bug on getGitInfo (common.js)

### ISSUE THREAD

**Issue by leopignataro** [dotnet/docfx#1886]
Dear DocFX team,

While debugging an issue with "contribute links" on my project, I stumbled upon something that I think might be a bug on the file [docfx.website.themes/common/common.js](https://github.com/dotnet/docfx/blob/dev/src/docfx.website.themes/common/common.js), function `getGitInfo`:

```javascript
function getGitInfo(gitContribute, gitRemote) {
    // apiSpecFolder defines the folder contains overwrite files for MRef, the default value is apiSpec
    var defaultApiSpecFolder = "apiSpec";

    if (!gitContribute && !gitRemote) {
        return {
            apiSpecFolder: defaultApiSpecFolder
        }
    }

    if (!gitContribute) {
        return {
            apiSpecFolder: defaultApiSpecFolder,
            repo: gitRemote.repo,
            branch: gitRemote.branch,
            // path defines the relative path from current git repo.
            path: gitRemote.path
        }
    }

    return {
        apiSpecFolder: gitContribute.apiSpecFolder || defaultApiSpecFolder,
        repo: gitContribute.repo || gitRemote.repo,
        branch: gitContribute.branch || gitRemote.branch,
        path: gitRemote.path
    }
}
```

This function results in error if the parameter `gitContribute` is filled but `gitRemote` is not (which is exactly what is happening on my project).

Shouldn't this

```javascript
if (!gitContribute && !gitRemote) {
```

Be this?

```javascript
if (!gitRemote) {
```

Since the last two return statements depend on `gitRemote` being defined?

**Comment by superyyrrzz** [comment]
@leopignataro Yes, it's a bug. I'll fix it.

**Comment by superyyrrzz** [comment]
This is fixed in v2.22
